### PR TITLE
Add versioning of the REST API

### PR DESCRIPTION
**Pull request by ewelinagr**

### CODE REVIEW

File: grails-app/conf/RestApiUrlMappings.groovy [hunk 1]
```diff
@@ -26,19 +26,20 @@
 class RestApiUrlMappings {
 
     // grails url-mappings-report can come handy here...
+    static String version = 'v1'
```
Comment (JanKanis) [inline grails-app/conf/RestApiUrlMappings.groovy]: Not sure if this should be in a variable, we want to be able to support multiple versions at the same time. So at least this shouldn't be named 'version'

File: test/functional/org/transmartproject/rest/ConceptsResourceTests.groovy [hunk 2]
```diff
@@ -134,20 +135,20 @@ class ConceptsResourceTests extends ResourceTestCase {
                 longConceptKey,
                 longConceptName,
                 longConceptPath,
-                longConceptUrl,
+                this.longConceptUrl,
```
Comment (forus) [inline test/functional/org/transmartproject/rest/ConceptsResourceTests.groovy]: Why `this` applied only partially.

File: grails-app/conf/RestApiUrlMappings.groovy [hunk 3]
```diff
@@ -26,19 +26,20 @@
...
 
     static mappings = {
-        '/studies'(controller: 'study', method: 'GET', resources: 'study', includes: ['index', 'show'])
+        "/$version/studies"(controller: 'study', method: 'GET', resources: 'study', includes: ['index', 'show'])
```
Comment (JanKanis) [inline grails-app/conf/RestApiUrlMappings.groovy]: Another way to factor this part out is to do something like

    '/v1' {
        '/studies'(...)
        ...
    }

I think that would be better in this case, then we can create a `/v2` block as well.

File: src/groovy/org/transmartproject/rest/marshallers/StudySerializationHelper.groovy [hunk 4]
```diff
@@ -36,12 +36,11 @@ import static grails.rest.render.util.AbstractLinkingRenderer.RELATIONSHIP_SELF
 class StudySerializationHelper extends AbstractHalOrJsonSerializationHelper<Study> {
 
     final Class targetType = Study
-
     final String collectionName = 'studies'
 
     @Override
     Collection<Link> getLinks(Study study) {
-        [new Link(RELATIONSHIP_SELF, '/studies/' +
+        [new Link(RELATIONSHIP_SELF, "/studies/" +
```
Comment (forus) [inline src/groovy/org/transmartproject/rest/marshallers/StudySerializationHelper.groovy]: Is this peace missing something?

**Comment by JanKanis** [comment]
Note: before merging it needs to be verified that an existing client continues to work if it's transmart url is changed from "http://somehost/transmart" to "http://somehost/transmart/v1".
There should probably be an accompanying PR for transmartApp to redirect the OAuth endpoints, I didn't see that yet.

**Comment by JanKanis** [comment]
The `apiVersionController` seems to be missing

**Comment by forus** [comment]
Wouldn't it be an idea to make one commit out of all 4? So there would be no commits where application does not work properly.

**Comment by JanKanis** [comment]
gb-dev should also be merged with master

**Pull request merged**